src/Components/Sidebar/SidebarOption.tsx

```diff
@@ -0,0 +1,18 @@
+import { boolean } from "yargs";
+import {SideBarOption} from "../../Types/types"
+type SideBarOptionType = {
+  option: SideBarOption;
+  isActive: boolean;
+  handlePageClick: (selectedPage: string ) => void
+}
+export default function SidebarOption({ option, isActive, handlePageClick }: SideBarOptionType) {
+  const classList = "sidebar-option" + (isActive ? " active" : "");
+  return (
+    <div className={classList} onClick={() => handlePageClick(option.name)}>
+      <div className="sidebar-icon">{option.icon != null && option.icon}</div>
+      <div>
+        {option.name && <label className="sidebar-label">{option.name}</label>}
```

Comment:
No need to use this condition.
`option.icon` is of type `JSX.Element` it will never be undefined | null.

Also be cautious while using && operator for conditional rendering.

Falsy expression will still cause the element after && to be skipped but will return the falsy expression. In the example below, `<div>0</div>` will be returned by the render method.

`const Component = () => {
const count = 0;
  return (
    <div>
      {count && <h1>Messages: {count}</h1>}
    </div>
  );
}`

Ref - https://legacy.reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator